subprojects/core/src/main/java/org/gradle/internal/build/BuildLayoutValidator.java

```diff
@@ -68,8 +70,8 @@ public void validate(StartParameterInternal startParameter) {
         StringBuilder message = new StringBuilder("Directory '");
         message.append(startParameter.getCurrentDir())
             .append("' does not contain a Gradle build.\n\n")
-            .append("A Gradle build should contain a 'settings.gradle' or 'settings.gradle.kts' file in its root directory. ")
-            .append("It may also contain a 'build.gradle' or 'build.gradle.kts' file.\n\n")
+            .append("A Gradle build should contain one of the possible settings files (").append(String.join(",", getValidSettingsFileNames())).append(") in its root directory. ")
```

Comment:
Minor suggestion: would the message read more clearly if rephrased as follows?

> A Gradle build's root directory should contain one of the possible settings files: ...

Also, should the separator be `", "` (with a space)?

Reply:
Ok, makes sense, adapting.